src/test/java/org/elasticsearch/index/query/TemplateQueryTest.java

```diff
@@ -372,6 +374,47 @@ public void testIndexedTemplate() throws Exception {
         assertHitCount(sr, 4);
     }
 
+    // Relates to #10397
+    @Test
+    public void testIndexedTemplateOverwrite() throws Exception {
+        createIndex("testindex");
+        ensureGreen("testindex");
+
+        index("testindex", "test", "1", jsonBuilder().startObject().field("searchtext", "dev1").endObject());
+        refresh();
+
+        for (int i = 1; i < randomIntBetween(2, 11); i++) {
```

Comment:
I don't think it is a good idea to call randomInBetween from here? We should save the result to a variable outside of the loop.